Tensile/KernelWriterConversion.py

```diff
@@ -236,11 +236,19 @@ def kernelBody(self):
       indexChar = self.indexChars[i]
       kStr += " + (size%s - 1) * strideW%s" % (indexChar, indexChar)
     kStr += ";" + self.endLine
-
+    
     kStr += "  " + self.datatype + " accum = 0;%s" % self.endLine
-    kStr += "  for (int i=0; i<gsu; i++) {%s" % self.endLine
+    kStr += "  " + self.datatype + " accum1 = 0;%s" % self.endLine
+    kStr += "  int res_gsu = gsu & 1;%s" % self.endLine
+    kStr += "  int gsu_end = gsu - res_gsu;%s" % self.endLine
+    kStr += "  for (int i=0; i<gsu_end; i+=2) {%s" % self.endLine
     kStr += "    accum += W[idxW];%s" % self.endLine
-    kStr += "    idxW  += strideW;%s" % self.endLine
+    kStr += "    accum1 += W[idxW+strideW];%s" % self.endLine
+    kStr += "    idxW  += 2*strideW;%s" % self.endLine
```

Comment:
As long as I checked, the compiler generates 2 * strideW inside for loop.
We can move 2 * strideW out of for loop by creating another variable like this.
   int strideW2 = strideW * 2;
   for ....

Reply:
Updated.

Reply:
Unrolling may help here. I am trying this change for other sizes. However, GSU is a small number when GSU algo is a MB (due to memory limitations). Thus, this for loop might not impose a huge overhead.